internal/handlers/common/update.go

```diff
@@ -110,6 +120,55 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 	return changed, nil
 }
 
+// processCurrentdateFieldExpression changes document according to $currentDate operator.
+// If the document was changed it returns true.
+func processCurrentdateFieldExpression(doc *types.Document, currentDateVal any) (bool, error) {
+	var changed bool
+	var err error
+	currentDateExpression := currentDateVal.(*types.Document)
+
+	now := time.Now()
+	sort.Strings(currentDateExpression.Keys())
+
+	for _, field := range currentDateExpression.Keys() {
+		currentDateField := must.NotFail(currentDateExpression.Get(field))
+
+		switch currentDateField := currentDateField.(type) {
+		case bool:
+			if err = doc.Set(field, now); err != nil {
+				return false, err
+			}
+			changed = true
+
+		case *types.Document:
+			currentDateType, err := currentDateField.Get("$type")
+			if err != nil { // default is date
+				if err := doc.Set(field, now); err != nil {
+					return false, err
+				}
+				changed = true
+				continue
+			}
+
+			currentDateType = currentDateType.(string)
+			switch currentDateType {
+			case "timestamp":
+				if err := doc.Set(field, types.NextTimestamp(uint64(now.UTC().Unix()))); err != nil {
```

Comment:
Oh, here we do some conversion to UTC, but we didn't do it for `doc.Set` above. Is it because the timestamp is always stored in UTC (I've never thought about it)?

Reply:
> Oh, here we do some conversion to UTC, but we didn't do it for `doc.Set` above. Is it because the timestamp is always stored in UTC (I've never thought about it)?

For `Date` it's explicitly specified that `Date` has to be in UTC. For timestamp - is not. But I guess as the operator is the same, It's also should be in UTC.